tsl/src/continuous_aggs/create.c

```diff
@@ -1487,6 +1487,13 @@ cagg_validate_query(const Query *query, const bool finalized, const char *cagg_s
 		}
 
 		ts_cache_release(hcache);
+
+		/*
+		 * We need a GROUP By clause with time_bucket on the partitioning
+		 * column of the hypertable
+		 */
+		Assert(query->groupClause);
+		caggtimebucket_validate(&bucket_info, query->groupClause, query->targetList);
```

Comment:
Since you're moving the code into an if-statement, it is a possibility that not setting `query->groupClause` `query->targetList` could cause issues later in the execution. I am not sure if that can be triggered in a reasonable way, but could you investigate this?